src/gen9_hevc_encoder.c

```diff
@@ -0,0 +1,7376 @@
+
+    gpe_context->vfe_desc7.scoreboard2.delta_x4 = 0;
+    gpe_context->vfe_desc7.scoreboard2.delta_y4 = -2;
+
+    gpe_context->vfe_desc7.scoreboard2.delta_x5 = 0;
+    gpe_context->vfe_desc7.scoreboard2.delta_y5 = -3;
+
+    gpe_context->vfe_desc7.scoreboard2.delta_x6 = 1;
+    gpe_context->vfe_desc7.scoreboard2.delta_y6 = -2;
+
+    gpe_context->vfe_desc7.scoreboard2.delta_x7 = 1;
+    gpe_context->vfe_desc7.scoreboard2.delta_y7 = -3;
+}
+
+static void
+gen9_hevc_vme_set_scoreboard_26(struct i965_gpe_context *gpe_context,
+                               unsigned int mask,
+                               unsigned int enable,
+                               unsigned int type)
+{
+    gpe_context->vfe_desc5.scoreboard0.mask = mask;
+    gpe_context->vfe_desc5.scoreboard0.type = type;
+    gpe_context->vfe_desc5.scoreboard0.enable = enable;
+
+    gpe_context->vfe_desc6.scoreboard1.delta_x0 = -1;
+    gpe_context->vfe_desc6.scoreboard1.delta_y0 = 0;
+
+    gpe_context->vfe_desc6.scoreboard1.delta_x1 = -1;
+    gpe_context->vfe_desc6.scoreboard1.delta_y1 = -1;
+
+    gpe_context->vfe_desc6.scoreboard1.delta_x2 = 0;
+    gpe_context->vfe_desc6.scoreboard1.delta_y2 = -1;
+
+    gpe_context->vfe_desc6.scoreboard1.delta_x3 = 1;
+    gpe_context->vfe_desc6.scoreboard1.delta_y3 = -1;
+}
+
+static void
+gen9_hevc_init_object_walker(struct hevc_enc_kernel_walker_parameter *hevc_walker_param,
+                             struct gpe_media_object_walker_parameter *gpe_param)
+{
+    memset(gpe_param, 0, sizeof(*gpe_param));
+
+    gpe_param->use_scoreboard = hevc_walker_param->use_scoreboard;
+    gpe_param->block_resolution.x = hevc_walker_param->resolution_x;
+    gpe_param->block_resolution.y = hevc_walker_param->resolution_y;
+    gpe_param->global_resolution.x = hevc_walker_param->resolution_x;
+    gpe_param->global_resolution.y = hevc_walker_param->resolution_y;
+    gpe_param->global_outer_loop_stride.x = hevc_walker_param->resolution_x;
+    gpe_param->global_outer_loop_stride.y = 0;
+    gpe_param->global_inner_loop_unit.x = 0;
+    gpe_param->global_inner_loop_unit.y = hevc_walker_param->resolution_y;
+    gpe_param->local_loop_exec_count = 0xFFFF;
+    gpe_param->global_loop_exec_count = 0xFFFF;
+
+    if (hevc_walker_param->no_dependency) {
+        gpe_param->scoreboard_mask = 0;
+        gpe_param->use_scoreboard = 0;
+        gpe_param->local_outer_loop_stride.x = 0;
+        gpe_param->local_outer_loop_stride.y = 1;
+        gpe_param->local_inner_loop_unit.x = 1;
+        gpe_param->local_inner_loop_unit.y = 0;
+        gpe_param->local_end.x = hevc_walker_param->resolution_x - 1;
+        gpe_param->local_end.y = 0;
+    }
+}
+
+static void
+gen9_hevc_init_object_walker_26z(struct gen9_hevc_encoder_state *priv_state,
+                                 struct i965_gpe_context *gpe_context,
+                                 struct gpe_media_object_walker_parameter *gpe_param,
+                                 int split_count,
+    
```

Comment:
It will be better to use the explicit definition for BTI definition. The usage of bti_idx++ is difficult to understand the meaning.